### PR TITLE
Modify -Xmn/-Xmns/-Xmnx documentation for balanced GC changes

### LINKED ISSUE #834: Modify/Enhance `-Xmn/-Xmns/-Xmnx` documentation
**Issue or pull request number:**
dependant on https://github.com/eclipse-openj9/openj9/pull/12043

**Overview:**
There are 2 modifications that should be made, WRT https://www.eclipse.org/openj9/docs/xmn/
- [x] these options also apply to balanced GC (true regardless of the PR mentioned above). In balanced GC, `nursery` is actually called `eden`, which plays a very very similar conceptual role (and obeys`-Xmn/-Xmns/-Xmnx` restrictions)
- [x] Once https://github.com/eclipse-openj9/openj9/pull/12043 has been merged, the default values for balanced GC will change. The default for `-Xmnx` will be 75% of `-Xmx`, while the default for `-Xmns` is simply 1 region. Specifying `-Xmn/-Xmns/-Xmnx` will overwrite the defaults (same behaviour as gencon in this case). Please note that modifying these default values may affect balanced GC's ability to satisfy `-Xgc:targetPausetime` (see https://github.com/eclipse-openj9/openj9-docs/issues/833) and `dnssExpectedRatioMaximum`/`dnssExpectedRatioMinimum`

**Release target:**
 Eclipse OpenJ9 0.30.0

**Applies to the following JDK versions:**
all

**Applies to the following platforms:**
all

**For new command line options:**
no new options - re-use `-Xmn/-Xmns/-Xmnx`

### PR DESCRIPTION
**Pull request by cedrichansen**
https://github.com/eclipse-openj9/openj9/pull/12043 will add support for the options mentioned above, for the Balanced GC policy.
This PR contains the documentation changes required to reflect this new behaviour.

closes #834

Signed-off-by: Cedric Hansen <cedric.hansen@ibm.com>

### CODE REVIEW

**Comment by cedrichansen** [comment]
@amicic @dmitripivkine FYI

**Comment by cedrichansen** [comment]
This PR will be ready to merge, once eclipse-openj9/openj9#12043 is also merged and released

**Comment by cedrichansen** [comment]
Jenkins doc stage

**Review by amicic — APPROVED** [review]

**Comment by pshipton** [comment]
Pls update the release notes as well, both for https://eclipse-openj9.github.io/openj9-docs/version0.30/ and in OpenJ9 https://github.com/eclipse-openj9/openj9/blob/master/doc/release-notes/0.30/0.30.md

**Comment by pshipton** [comment]
jenkins doc stage

**Comment by pshipton** [comment]
https://eclipse-openj9.github.io/openj9-docs-staging/841

File: docs/xmn.md
Comment (pshipton) [inline docs/xmn.md:48]: The links here don't work, I expect because of the `##` instead of `#`.

**Comment by pshipton** [comment]
jenkins doc stage

**Review by pshipton — APPROVED** [review]

**Pull request merged**
